### PR TITLE
Added calculation of lower/upper bound for divide operation

### PR DESCRIPTION
**Pull request by sadolini**
Root cause analysis: on Wenet model shape inference failed with divide on zero exception for evaluation of bounds for dynamic shapes

Solution: added calculation of bounds with detection 0s in the second argument and return max_value for such cases.

Ticket: 68487

### CODE REVIEW

File: ngraph/core/include/openvino/op/divide.hpp [hunk 1]
```diff
@@ -49,6 +49,10 @@ class OPENVINO_API Divide : public util::BinaryElementwiseArithmetic {
     bool evaluate(const HostTensorVector& outputs, const HostTensorVector& inputs) const override;
     OPENVINO_SUPPRESS_DEPRECATED_END
     bool has_evaluate() const override;
+    OPENVINO_SUPPRESS_DEPRECATED_START
+    bool evaluate_lower(const HostTensorVector& outputs) const override;
+    bool evaluate_upper(const HostTensorVector& outputs) const override;
+    OPENVINO_SUPPRESS_DEPRECATED_END
```
Comment (ilya-lavrenov) [inline ngraph/core/include/openvino/op/divide.hpp:55]: Can we add evaluate methods with runtime tensors instead?

Cc @ilyachur

Comment (sadolini) [reply ngraph/core/include/openvino/op/divide.hpp:55]: Inside these functions I use evaluate methods of several other operations with HostTensorVector. If I will use ov::runtime::TensorVector, I will need convert it to HostTensorVector and back similar to how it is done in node.cpp. 

@ilya-lavrenov @ilyachur Does it make sense in this case?

Comment (sadolini) [reply ngraph/core/include/openvino/op/divide.hpp:55]: I call inside these methods several other evaluate methods for different operations with HostTensorVector. So, I will need to convert ov::runtime::TensorVector to HostTensorVector and back as it was done in node.cpp.

@ilya-lavrenov @ilyachur Does it make sense to change these functions now?

Comment (ilyachur) [reply ngraph/core/include/openvino/op/divide.hpp:55]: In general it already should work from the box

Comment (sadolini) [reply ngraph/core/include/openvino/op/divide.hpp:55]: Evaluate methods with RuntimeTensors requires will automatically workable due to conversion in node.cpp, so I will not re-write code to use TensorVector. Almost all other operations have evaluate methods with HostTensorVector and infrastructure functions like NGRAPH_CHECK, creating tensor from constant works with HostTensor.

File: ngraph/core/src/op/divide.cpp [hunk 2]
```diff
@@ -105,3 +157,11 @@ bool op::v1::Divide::has_evaluate() const {
...
     return false;
 }
+
+bool ov::op::v1::Divide::evaluate_lower(const HostTensorVector& outputs) const {
+    return divide::evaluate_bound(this, outputs, false);
+}
+
+bool ov::op::v1::Divide::evaluate_upper(const HostTensorVector& outputs) const {
+    return divide::evaluate_bound(this, outputs, true);
+}
```
Comment (achetver) [inline ngraph/core/src/op/divide.cpp:167]: nit: add newline at the end of file

**Comment by achetver** [comment]
In general looks good. Do we need to add any tests for new functions?

**Review by vgavrilo — APPROVED** [review]

**Comment by sadolini** [comment]
> In general looks good. Do we need to add any tests for new functions?

Added test for failed on Wenet case. Also this functionality covered by ngraph/type_prop tests.

**Review by achetver — APPROVED** [review]

File: model-optimizer/mo/back/offline_transformations.py [hunk 3]
```diff
@@ -72,8 +72,9 @@ def read_model(path_to_xml):
     parser = argparse.ArgumentParser()
     parser.add_argument("--input_model")
     parser.add_argument("--framework")
-    parser.add_argument("--transform")
+    parser.add_argument("--transform", nargs='?', const='')
```
Comment (jane-intel) [inline model-optimizer/mo/back/offline_transformations.py:75]: How's that relevant?

**Review by jane-intel — APPROVED** [review]
This PR would use more tests. Please add as a separate PR in case this is critical

File: model-optimizer/mo/back/offline_transformations.py (hunk 3 above)
Comment (sadolini) [reply model-optimizer/mo/back/offline_transformations.py:75]: It's not relevant, anyway it is already not needed, will remove during rebase on latest master

**Pull request merged**